sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/ConfluentSchemaRegistryDeserializerProvider.java

```diff
@@ -73,25 +73,29 @@
   }
 
   public static <T> ConfluentSchemaRegistryDeserializerProvider<T> of(
-      String schemaRegistryUrl, String subject) {
-    return of(schemaRegistryUrl, subject, null, null);
+      String schemaRegistryUrl, Integer schemaRegistryCacheCapacity, String subject) {
+    return of(schemaRegistryUrl, schemaRegistryCacheCapacity, subject, null, null);
   }
 
   public static <T> ConfluentSchemaRegistryDeserializerProvider<T> of(
-      String schemaRegistryUrl, String subject, @Nullable Integer version) {
-    return of(schemaRegistryUrl, subject, version, null);
+      String schemaRegistryUrl,
+      Integer schemaRegistryCacheCapacity,
+      String subject,
+      @Nullable Integer version) {
+    return of(schemaRegistryUrl, schemaRegistryCacheCapacity, subject, version, null);
   }
 
   public static <T> ConfluentSchemaRegistryDeserializerProvider<T> of(
       String schemaRegistryUrl,
+      Integer schemaRegistryCacheCapacity,
       String subject,
       @Nullable Integer version,
       @Nullable Map<String, ?> schemaRegistryConfigs) {
     return new ConfluentSchemaRegistryDeserializerProvider(
         (SerializableFunction<Void, SchemaRegistryClient>)
             input ->
                 new CachedSchemaRegistryClient(
-                    schemaRegistryUrl, Integer.MAX_VALUE, schemaRegistryConfigs),
```

Comment:
We try to keep backwards compatibility when possible so it would be nice to keep the existing methods around even if marked experimental.

Finally, we should change this to `1000` since that seems to be the default if _unspecified_.